extensions/blocks/map/map.php

```diff
@@ -25,7 +25,98 @@
 function jetpack_map_block_load_assets( $attr, $content ) {
 	$api_key = Jetpack_Options::get_option( 'mapbox_api_key' );
 
+	if ( class_exists( 'Jetpack_AMP_Support' ) && Jetpack_AMP_Support::is_amp_request() ) {
+		static $map_block_counter = [];
+
+		$id = get_the_ID();
+		if ( ! $map_block_counter[ $id ] ) {
+			$map_block_counter[ $id ] = 0;
+		}
+		$map_block_counter[ $id ]++;
+
+		$iframe_url = add_query_arg(
+			array(
+				'map-block-counter' => $map_block_counter[ get_the_ID() ],
+				'map-block-post-id' => get_the_ID(),
+			),
+			get_permalink()
+		);
+
+		$placeholder = preg_replace( '/(?<=<div\s)/', 'placeholder ', $content );
+
+		return sprintf(
+			'<amp-iframe src="%s" width="%d" height="%d" layout="responsive" allowfullscreen sandbox="allow-scripts">%s</amp-iframe>',
+			esc_url( $iframe_url ),
+			4,
+			3,
+			$placeholder
+		);
+	}
+
 	Jetpack_Gutenberg::load_assets_as_required( 'map' );
 
 	return preg_replace( '/<div /', '<div data-api-key="' . esc_attr( $api_key ) . '" ', $content, 1 );
 }
+
+/**
+ * Render a page containing only a single Map block.
+ */
+function jetpack_map_block_render_single_block_page() {
+	// phpcs:ignore WordPress.Security.NonceVerification
+	$map_block_counter = isset( $_GET, $_GET['map-block-counter'] ) ? absint( $_GET['map-block-counter'] ) : null;
+	// phpcs:ignore WordPress.Security.NonceVerification
+	$map_block_post_id = isset( $_GET, $_GET['map-block-post-id'] ) ? absint( $_GET['map-block-post-id'] ) : null;
+
+	if ( ! $map_block_counter || ! $map_block_post_id ) {
+		return;
+	}
+
+	/* Create an array of all root-level DIVs that are Map Blocks */
+	$post = get_post( $map_block_post_id );
+
+	if ( ! class_exists( 'DOMDocument' ) ) {
+		return;
+	}
+
+	$post_html = new DOMDocument();
+	$content   = apply_filters( 'the_content', $post->post_content );
+
+	/* Suppress warnings */
+	libxml_use_internal_errors( true );
+	$post_html->loadHTML( $content );
```

Comment:
In the past, we've done this as well after feedback from users:

```suggestion
	@$post_html->loadHTML( $content ); // phpcs:ignore WordPress.PHP.NoSilencedErrors.Discouraged
```

Reply:
Done: https://github.com/Automattic/jetpack/pull/13405/commits/d9492d1aacd8bcc271d3fb0c1a6f85e6663d87f3